examples/transactions/examples/encode_and_decode_raw_transaction.rs

```diff
@@ -0,0 +1,102 @@
+//! Example of encoding and decoding raw transactions.
+
+use alloy::primitives::keccak256;
+use alloy::primitives::private::alloy_rlp::Decodable;
+use alloy::primitives::private::alloy_rlp::Encodable;
+use alloy::primitives::FixedBytes;
+use alloy::primitives::TxKind;
+use alloy::primitives::U256;
+use alloy::providers::WalletProvider;
+use alloy::{
+    consensus::TxEnvelope,
+    consensus::{SignableTransaction, TxEip1559},
+    hex,
+    network::{EthereumWallet, TransactionBuilder},
+    primitives::Address,
+    providers::{Provider, ProviderBuilder},
+    rpc::types::TransactionRequest,
+};
+use eyre::Result;
+
+fn build_unsigned_tx(chain_id: u64, to_address: Address) -> TxEip1559 {
+    let mut tx = TxEip1559::default();
+    tx.chain_id = chain_id;
+    tx.nonce = 0;
+    tx.gas_limit = 21_000;
+    tx.max_fee_per_gas = 20_000_000_000;
+    tx.max_priority_fee_per_gas = 1_000_000_000;
+    tx.to = TxKind::Call(to_address); // Change this to `TxKind::Create` if you'd like to deploy a contract instead
+    tx.value = U256::from(100);
+
+    tx
+}
+
+fn unsigned_tx_to_bytes(tx: TxEip1559) -> Vec<u8> {
+    tx.encoded_for_signing() // To use this, have to import "alloy::primitives::private::alloy_rlp::Encodable"
+}
+
+fn bytes_to_unsigned_tx(bytes: Vec<u8>) -> TxEip1559 {
+    let mut slice = &bytes.as_slice()[1..];
+    TxEip1559::decode(&mut slice).unwrap() // To use this, have to import "alloy::primitives::private::alloy_rlp::Decodable"
+}
+
+async fn sign_tx(tx: TxEip1559, wallet: EthereumWallet) -> TxEnvelope {
+    let tx_request: TransactionRequest = tx.into();
+
+    tx_request.build(&wallet).await.unwrap()
+}
+
+fn signed_tx_to_bytes(signed_tx: TxEnvelope) -> Vec<u8> {
+    let mut encoded = Vec::new();
+    signed_tx.encode(&mut encoded);
+    let encoded = &encoded[2..];
+    encoded.into()
+}
+
+fn bytes_to_signed_tx(bytes: Vec<u8>) -> TxEnvelope {
+    let mut slice = bytes.as_slice();
+    TxEnvelope::decode(&mut slice).unwrap()
+}
+
+fn get_tx_hash_from_signed_tx_bytes(signed_tx_bytes: Vec<u8>) -> FixedBytes<32> {
+    format!("0x{}", hex::encode(keccak256(signed_tx_bytes.clone())))
+        .parse::<FixedBytes<32>>()
+        .unwrap()
+}
```

Comment:
I think we can use the `hex!` macro here

Reply:
Ok. I removed this function, and just use `keccak256()` only